scene/3d/node_3d.cpp

```diff
@@ -85,12 +85,20 @@ void Node3D::_notify_dirty() {
 }
 
 void Node3D::_update_local_transform() const {
-	if (this->get_rotation_edit_mode() != ROTATION_EDIT_MODE_BASIS) {
-		data.local_transform = data.local_transform.orthogonalized();
-	}
-	data.local_transform.basis.set_euler_scale(data.rotation, data.scale);
+	// This function is called when the local transform (data.local_transform) is dirty and the right value is contained in the euler vector
```

Comment:
Nitpicks but while at it, let's get comment style right/consistent:
```suggestion
	// This function is called when the local transform (data.local_transform) is dirty and the right value is contained in the Euler vector.
```